pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -259,81 +301,60 @@ func (c *Controller) syncHandler(key string) error {
-	//	runtime.HandleError(fmt.Errorf("%s: deployment name must be specified", key))
-	//	return nil
-	//}
-	//
-	//// Get the deployment with the name specified in EtcdStorage.spec
-	//deployment, err := c.deploymentsLister.Deployments(etcdstorage.Namespace).Get(deploymentName)
-	//// If the resource doesn't exist, we'll create it
-	//if errors.IsNotFound(err) {
-	//	deployment, err = c.kubeclientset.AppsV1().Deployments(etcdstorage.Namespace).Create(newDeployment(etcdstorage))
-	//}
-	//
-	//// If an error occurs during Get/Create, we'll requeue the item so we can
-	//// attempt processing again later. This could have been caused by a
-	//// temporary network failure, or any other transient reason.
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//// If the Deployment is not controlled by this EtcdStorage resource, we should log
-	//// a warning to the event recorder and ret
-	//if !metav1.IsControlledBy(deployment, etcdstorage) {
-	//	msg := fmt.Sprintf(MessageResourceExists, deployment.Name)
-	//	c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceExists, msg)
-	//	return fmt.Errorf(msg)
-	//}
-	//
-	//// If this number of the replicas on the EtcdStorage resource is specified, and the
-	//// number does not equal the current desired replicas on the Deployment, we
-	//// should update the Deployment resource.
-	//if etcdstorage.Spec.Replicas != nil && *etcdstorage.Spec.Replicas != *deployment.Spec.Replicas {
-	//	glog.V(4).Infof("EtcdStorage %s replicas: %d, deployment replicas: %d", name, *etcdstorage.Spec.Replicas, *deployment.Spec.Replicas)
-	//	deployment, err = c.kubeclientset.AppsV1().Deployments(etcdstorage.Namespace).Update(newDeployment(etcdstorage))
-	//}
-	//
-	//// If an error occurs during Update, we'll requeue the item so we can
-	//// attempt processing again later. THis could have been caused by a
-	//// temporary network failure, or any other transient reason.
-	//if err != nil {
-	//	return err
-	//}
-
-	//// Finally, we update the status block of the EtcdStorage resource to reflect the
-	//// current state of the world
-	//err = c.updateEtcdStorageStatus(etcdstorage, deployment)
-	//if err != nil {
-	//	return err
-	//}
+	replicaset, err := c.replicasetsLister.ReplicaSets(c.controllerNamespace).Get(getReplicaSetName(etcdstorage))
+	if errors.IsNotFound(err) {
+		replicaset, err = c.kubeclientset.AppsV1().ReplicaSets(c.controllerNamespace).Create(c.newReplicaSet(etcdstorage))
+	}
+
+	// If an error occurs during Get/Create, we'll requeue the item so we can
+	// attempt processing again later. This could have been caused by a
+	// temporary network failure, or any other transient reason.
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+
+	// If the ReplicaSet is not controlled by this EtcdStorage resource, we should log
+	// a warning to the event recorder and ret
+	if !metav1.IsControlledBy(replicaset, etcdstorage) {
```

Comment:
why don't we just ignore these replicasets? Why a warning? If there is another replicaset in the namespace, who cares?

Reply:
oh, nvmd. Mixed the func up. Here we know that it is the replicaset with the given name.